hooks/terraform_validate.sh

```diff
@@ -43,16 +80,64 @@ function per_dir_hook_unique_part {
   local -a -r args=("$@")
 
   local exit_code
-  local validate_output
+  #
+  # Get hook settings
+  #
+  local retry_once_with_cleanup=false
+
+  IFS=";" read -r -a configs <<< "${HOOK_CONFIG[*]}"
+
+  for c in "${configs[@]}"; do
+
+    IFS="=" read -r -a config <<< "$c"
+    key=${config[0]}
+    value=${config[1]}
+
+    case $key in
+      --retry-once-with-cleanup)
+        retry_once_with_cleanup=$value
+        ;;
+    esac
+  done
 
   common::terraform_init 'terraform validate' "$dir_path" || {
     exit_code=$?
     return $exit_code
   }
 
-  # pass the arguments to hook
-  validate_output=$(terraform validate "${args[@]}" 2>&1)
-  exit_code=$?
+  if [ "$retry_once_with_cleanup" != "true" ]; then
+    # terraform validate only
+    validate_output=$(terraform validate "${args[@]}" 2>&1)
+    exit_code=$?
+  else
+    # terraform validate, plus capture possible errors
+    validate_output=$(terraform validate -json "${args[@]}" 2>&1)
+    exit_code=$?
+
+    # Match specific validation errors
+    local -i validate_errors_matched
+    match_validate_errors "$validate_output"
+    validate_errors_matched=$?
+
+    # Errors matched; Retry validation
+    if [ "$validate_errors_matched" = "1" ]; then
```

Comment:
Arithmetic evaluation (comparison of integers in this case) should use `-eq` instead of `=` (or `==`, which is the same as `=`):
```suggestion
    if [ "$validate_errors_matched" -eq 1 ]; then
```